pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4341,7 +4344,7 @@ def _get_offset(name: str) -> BaseOffset:
     return _offset_map[name]
 
 
-cpdef to_offset(freq):
+cpdef to_offset(freq, bint is_period=False):
```

Comment:
Since `to_offset` is a public function, I'm not sure if it makes sense for users to need to specify/care about `is_period`?

Reply:
in order to clarify the behaviour of `to_offset `we can later add new examples in documentation, like examples below

```
ser = pd.Series([1, 2], index=pd.DatetimeIndex(['2023-01-01', '2023-02-01']))
ser.resample(to_offset("2ME")).sum()
```
and
```
pd.PeriodIndex([pd.Period("2013-01", to_offset("2M", is_period=True))])
```

Reply:
sure but would people use `to_offset` this way?

I think just showing
```python
>>> to_offset('ME')
<MonthEnd>
>>> to_offset('M', is_period=True)
 <MonthEnd>
```
should be OK

in any case, although technically public, I doubt many people are using this function directly